include/boost/math/special_functions/prime_sieve.hpp

```diff
@@ -0,0 +1,402 @@
+// Copyright 2020 Matt Borland
+//
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SPECIAL_FUNCTIONS_PRIME_SIEVE_HPP
+#define BOOST_MATH_SPECIAL_FUNCTIONS_PRIME_SIEVE_HPP
+
+#include <boost/math/special_functions/prime.hpp>
+#include <boost/assert.hpp>
+#include <vector>
+#include <iterator>
+#include <cmath>
+#include <thread>
+#include <memory>
+#include <future>
+
+#ifdef _MSVC_LANG
+#if _MSVC_LANG >= 201703 // _MSVC_LANG == __cplusplus: https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
+#include <execution>
+#endif
+#else
+#if __cplusplus >= 201703
+#include <execution>
+#endif
+#endif
+
+namespace boost { namespace math { namespace detail
+{
+// https://mathworld.wolfram.com/SieveofEratosthenes.html
+// https://www.cs.utexas.edu/users/misra/scannedPdf.dir/linearSieve.pdf
+template<class Z, class Container>
+void linear_sieve(Z upper_bound, Container &c)
```

Comment:
Please use something more descriptive than `c` here. :) Surely the values in this container have some meaning, especially since it's the prime numbers result, right?